code/modules/surgery/robotics.dm

```diff
@@ -666,3 +666,44 @@
 	user.visible_message("<span class='warning'> [user]'s [tool.name] slips, failing to reprogram [target]'s [affected.name].</span>",
 	"<span class='warning'> Your [tool.name] slips, failing to reprogram [target]'s [affected.name].</span>")
 	return SURGERY_STEP_RETRY
+
+/datum/surgery/robotics/reconfigure_id
+	name = "Reconfigure Identity"
+	steps = list(
+		/datum/surgery_step/robotics/external/unscrew_hatch,
+		/datum/surgery_step/robotics/external/open_hatch,
+		/datum/surgery_step/robotics/edit_serial,
+		/datum/surgery_step/robotics/external/close_hatch)
+	possible_locs = list(BODY_ZONE_CHEST, BODY_ZONE_HEAD)
+
+/datum/surgery_step/robotics/edit_serial
+	name = "edit serial number"
+	allowed_tools = list(TOOL_MULTITOOL = 100)
+	time = 4.8 SECONDS
+
+/datum/surgery_step/robotics/edit_serial/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	user.visible_message(
+		"[user] begins to edit [target]'s identity parameters with [tool].",
+		"You begin to alter [target]'s identity parameters with [tool]."
+		)
+	return ..()
+
+/datum/surgery_step/robotics/edit_serial/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	var/old_name = target.real_name
+	var/new_name = copytext(reject_bad_text(input(user,"Choose a name for this machine.","Set Name","[old_name]") as null|text),1,MAX_NAME_LEN)
```

Comment:
You probably still want to move this input out to a variable and nullcheck it before you call `reject_bad_text` and `copytext` on it, I don't think those functions would respond nicely to a null input

Reply:
This is the one change I don't think is exactly necessary. `reject_bad_text` reads the input and outputs null if the input is empty or if it has bad characters, and if `copytext` couldn't handle a null input then it would break when I intentionally leave the input box blank during testing. If I split this line, I would also need to run the null check twice.

For extra context, I cannibalized this code from the hand labeler.

Reply:
Fair enough, as long as it works!